tests/run-make/reset-codegen-1/rmake.rs

```diff
@@ -31,8 +30,6 @@ fn main() {
         ("multi-output", Some("asm,obj")),
     ];
     for (output_file, emit) in flags {
-        fs::remove_file(output_file).unwrap_or_default();
-        compile(output_file, emit);
-        fs::remove_file(output_file);
+        compile(&bin_name(output_file), emit);
     }
```

Comment:
Concern: I would like to see the `remove_file`s remain because they also check the output filenames with no `--emit`, single-artifact `--emit` and multi-artifact `--emit`s. See [`8d182e3` (#126197)](https://github.com/rust-lang/rust/pull/126197/commits/8d182e3775817f0c5577079761e93c3b778280d6).

Reply:
This was my change, sorry. The previous code was highly non-obvious (I suspect that the removes were there because previously the code wasn't run in a temporary directory). I'd suggest to do this:
1) Explicitly add the expected paths as a third tuple item in the `flags` array
2) Add something like `assert!(is_file(expected_filename))` for each of the expected filenames. That makes it much more obvious than removing the file.

Reply:
Thanks, yeah, that's a much better way to doing this.